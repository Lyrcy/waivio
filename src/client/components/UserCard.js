import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import FollowButton from '../widgets/FollowButton';
import Avatar from '../components/Avatar';

import './UserCard.less';

const UserCard = ({ user, alt, showFollow }) => (
  <div className="UserCard">
    <div className="UserCard__left">
      <div className="UserCard__wrap">
        <Link to={`/@${user.name}`}>
          <Avatar username={user.name} size={40} />
        </Link>
        <Link to={`/@${user.name}`}>
          <span className="username">{user.name}</span>
        </Link>
      </div>
      {alt && <span className={showFollow ? 'UserCard__alt' : 'UserCard__short'}>{alt}</span>}
    </div>
    {showFollow && <FollowButton following={user.name} followingType="user" secondary />}
  </div>
);

UserCard.propTypes = {
  user: PropTypes.shape(),
  alt: PropTypes.node,
  showFollow: PropTypes.bool,
};

UserCard.defaultProps = {
  alt: '',
  user: {},
  showFollow: true,
  authUser: '',
};

export default UserCard;
