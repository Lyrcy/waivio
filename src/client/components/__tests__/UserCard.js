import React from 'react';
import { act } from 'react-dom/test-utils';
import { shallow, mount } from 'enzyme';

import { IntlProvider } from 'react-intl';
import UserCard from '../UserCard';

describe('<UserCard />', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      showFollow: true,
      user: '',
      alt: '',
    };

    wrapper = mount(
      <IntlProvider locale="en">
        <UserCard {...props} />
      </IntlProvider>,
    );
    act(() => {
      wrapper.update();
    });
  });

  it('should render without exploding', () => {
    wrapper = shallow(
      <IntlProvider locale="en">
        <UserCard {...props} />
      </IntlProvider>,
    );
    act(() => {
      wrapper.update();
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('Should include showFollow as in prop', () => {
    expect(wrapper.props().className).toBe('showFollow');
  });

  it('Should have defaultProps', () => {
    const defaultProps = {
      alt: '',
      user: {},
      showFollow: true,
      authUser: '',
    };

    wrapper = shallow(
      <IntlProvider locale="en">
        <UserCard {...props} />
      </IntlProvider>,
    );
    act(() => {
      wrapper.update();
    });
    expect(wrapper.props().children.type.defaultProps).toEqual(defaultProps);
  });

  it('Should include UserCard in className', () => {
    wrapper = mount(
      <IntlProvider locale="en">
        <UserCard {...props} />
      </IntlProvider>,
    );
    act(() => {
      wrapper.update();
    });
    expect(wrapper.find('div').props().className).toBe('UserCard');
  });
});
