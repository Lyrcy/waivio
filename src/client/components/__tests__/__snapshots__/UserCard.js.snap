// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCard /> should render without exploding 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserCard
    alt=""
    authUser=""
    showFollow={true}
    user=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="UserCard__left"
        >
          <div
            className="UserCard__wrap"
          >
            <UNDEFINED
              to="/@undefined"
            >
              <Avatar
                size={40}
                username={undefined}
              />
            </UNDEFINED>
            <UNDEFINED
              to="/@undefined"
            >
              <span
                className="username"
              />
            </UNDEFINED>
          </div>
          
        </div>,
        <UNDEFINED
          following={undefined}
          followingType="user"
          secondary={true}
        />,
      ],
      "className": "UserCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="UserCard__wrap"
            >
              <UNDEFINED
                to="/@undefined"
              >
                <Avatar
                  size={40}
                  username={undefined}
                />
              </UNDEFINED>
              <UNDEFINED
                to="/@undefined"
              >
                <span
                  className="username"
                />
              </UNDEFINED>
            </div>,
            "",
          ],
          "className": "UserCard__left",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <UNDEFINED
                  to="/@undefined"
                >
                  <Avatar
                    size={40}
                    username={undefined}
                  />
                </UNDEFINED>,
                <UNDEFINED
                  to="/@undefined"
                >
                  <span
                    className="username"
                  />
                </UNDEFINED>,
              ],
              "className": "UserCard__wrap",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Avatar
                    size={40}
                    username={undefined}
                  />,
                  "to": "/@undefined",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "size": 40,
                    "username": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "Link",
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <span
                    className="username"
                  />,
                  "to": "/@undefined",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "username",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "Link",
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                },
              },
            ],
            "type": "div",
          },
          "",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "following": undefined,
          "followingType": "user",
          "secondary": true,
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "withAuthActions(Connect(FollowButton))",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="UserCard__left"
          >
            <div
              className="UserCard__wrap"
            >
              <UNDEFINED
                to="/@undefined"
              >
                <Avatar
                  size={40}
                  username={undefined}
                />
              </UNDEFINED>
              <UNDEFINED
                to="/@undefined"
              >
                <span
                  className="username"
                />
              </UNDEFINED>
            </div>
            
          </div>,
          <UNDEFINED
            following={undefined}
            followingType="user"
            secondary={true}
          />,
        ],
        "className": "UserCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="UserCard__wrap"
              >
                <UNDEFINED
                  to="/@undefined"
                >
                  <Avatar
                    size={40}
                    username={undefined}
                  />
                </UNDEFINED>
                <UNDEFINED
                  to="/@undefined"
                >
                  <span
                    className="username"
                  />
                </UNDEFINED>
              </div>,
              "",
            ],
            "className": "UserCard__left",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <UNDEFINED
                    to="/@undefined"
                  >
                    <Avatar
                      size={40}
                      username={undefined}
                    />
                  </UNDEFINED>,
                  <UNDEFINED
                    to="/@undefined"
                  >
                    <span
                      className="username"
                    />
                  </UNDEFINED>,
                ],
                "className": "UserCard__wrap",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <Avatar
                      size={40}
                      username={undefined}
                    />,
                    "to": "/@undefined",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "size": 40,
                      "username": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "Link",
                    "propTypes": Object {
                      "innerRef": [Function],
                      "onClick": [Function],
                      "replace": [Function],
                      "target": [Function],
                      "to": [Function],
                    },
                    "render": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <span
                      className="username"
                    />,
                    "to": "/@undefined",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "username",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "Link",
                    "propTypes": Object {
                      "innerRef": [Function],
                      "onClick": [Function],
                      "replace": [Function],
                      "target": [Function],
                      "to": [Function],
                    },
                    "render": [Function],
                  },
                },
              ],
              "type": "div",
            },
            "",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "following": undefined,
            "followingType": "user",
            "secondary": true,
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "withAuthActions(Connect(FollowButton))",
            "type": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
